pages/index.tsx

```diff
@@ -1,8 +1,9 @@
 import React from 'react';
 import type { NextPage } from 'next';
+import Layout from '../components/common/layout';
 
 const Home: NextPage = () => {
-    return (
+    return <Layout pageTitle="Home">
```

Comment:
While I liked the idea of having Layout be an abstraction pages didn't need to be aware of, I wasn't a fan of trying to add some abstraction separated from the page implementation which would attempt to deduce the page title via other means. I considered doing an href lookup in a dictionary that stored page names or extending the route definition, but in the end, conventions like this are often not straight forward to understand.

So moving the layout component usage to within each page, and allowing the page to dictate the title, seems easier to understand.